blockchain/blockindex.go

```diff
@@ -16,6 +16,24 @@ import (
 	"github.com/btcsuite/btcd/wire"
 )
 
+const (
+	// statusDataStored indicates that the block's payload is stored on disk.
+	statusDataStored = 1 << iota
```

Comment:
The first one should specify the type so that `iota` will carry it through.  Otherwise these will all be untyped constants.

```Go
statusDataStored blockStatus = 1 << iota
statusValid
...
```

Also, please define the type before the `const` block (which will use it) for consistency.